src/Features/Core/Portable/UnusedReferences/ProjectAssets/ProjectAssetsReader.cs

```diff
@@ -72,18 +49,25 @@ internal static class ProjectAssetsReader
                 .ToImmutableHashSet();
             autoReferences ??= ImmutableHashSet<string>.Empty;
 
+            var targetLibraryKeys = projectAssets.Targets
+                .ToImmutableDictionary(t => t.Key, t => t.Value.ToImmutableDictionary(l => l.Key.Split('/')[0], l => l.Key));
```

Comment:
this def needs a comment.  plus, just blindly splitting/indexing scares me.  how do we know this is safe?

Reply:
I'll add a comment to the effect of "Targets contain a hashmap of Libraries keyed by `{LibraryName}/{LibraryVersion}` we need to split these keys and create a mapping of LibraryName to the complete library key."